### PR TITLE
Add a new section for general APB considerations

### PR DESCRIPTION
**Pull request by david-martin**
Adds 3 general considerations for apb development:
* scaling
* labelling
* passwords

Some header indent cleanup too

@maleck13 @aidenkeating @pb82 would you mind reviewing?

### CODE REVIEW

File: docs/walkthroughs/developing-apbs-locally.adoc [hunk 1]
```diff
@@ -46,11 +44,48 @@ apb init
 
 NOTE: Best practice is to name the APB ending with `-apb`.
 
-== Making your APB consumable as a mobile service
+== General APB considerations
+
+=== Scaling
+
+Use a link:https://kubernetes.io/docs/concepts/workloads/controllers/deployment/[Deployment] with a link:https://kubernetes.io/docs/concepts/workloads/controllers/replicaset/[ReplicaSet] for containers that can scale horizontally. This will allow users of the APB to scale the service if needed.
```
Comment (pb82) [inline docs/walkthroughs/developing-apbs-locally.adoc]: Do you really need to know about the ReplicaSet? As far as i know it's sufficient to use the `replicas` property of a `Deployment` and your replicas will be managed automatically: https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#replicas

Comment (david-martin) [reply docs/walkthroughs/developing-apbs-locally.adoc]: Fair point. I'll reword it to mention just the `replicas` property as thats sufficient to mention here.

Comment (david-martin) [reply docs/walkthroughs/developing-apbs-locally.adoc]: done

**Comment by pb82** [comment]
Looks good 👍

**Comment by AdamSaleh** [comment]
@david-martin managed to update the golden-files, so that the integration-tests are passing.
